### PR TITLE
Database checkpoints becomes inconsistent when running out of disk space

### PR DESCRIPTION
**Pull request by shaan1337**
Fixed: Database checkpoints becomes inconsistent when running out of disk space

Fixes: https://github.com/EventStore/home/issues/387
Fixes: https://github.com/EventStore/EventStore/issues/3452
Fixes: https://github.com/EventStore/EventStore/issues/3642

### CODE REVIEW

File: src/EventStore.Core/TransactionLog/Chunks/TFChunk/TFChunk.cs [hunk 1]
```diff
@@ -930,10 +930,11 @@ public enum ChunkVersions : byte {
...
 		}
 
-		public void Dispose() {
+		public void Dispose() => TryClose();
+
+		public bool TryClose() {
 			_selfdestructin54321 = true;
-			TryDestructFileStreams();
-			TryDestructMemStreams();
+			return TryDestructFileStreams() && TryDestructMemStreams();
```
Comment (timothycoleman) [inline src/EventStore.Core/TransactionLog/Chunks/TFChunk/TFChunk.cs:937]: Intentional short circuit?

Comment (shaan1337) [reply src/EventStore.Core/TransactionLog/Chunks/TFChunk/TFChunk.cs:937]: :man_facepalming: well spotted, thanks will fix it!

Comment (shaan1337) [reply src/EventStore.Core/TransactionLog/Chunks/TFChunk/TFChunk.cs]: fixed: 5ede56ca6e117528881b36152a118b344432d62d

**Comment by timothycoleman** [comment]
Nice, could consider squashing that commit in and then I think we're good to go :+1:

**Review by hayley-jean — APPROVED** [review]

**Review by timothycoleman — APPROVED** [review]
:+1:

**Pull request merged**